test/integration/statefulset/statefulset_test.go

```diff
@@ -479,3 +481,90 @@ func TestAutodeleteOwnerRefs(t *testing.T) {
 		})
 	}
 }
+
+func TestStatefulSetStartOrdinal(t *testing.T) {
+	tests := []struct {
+		ordinals         *appsv1.StatefulSetOrdinals
+		name             string
+		replicas         int
+		expectedPodNames []string
+	}{
+		{
+			name:             "default start ordinal, no ordinals set",
+			replicas:         3,
+			expectedPodNames: []string{"sts-0", "sts-1", "sts-2"},
+		},
+		{
+			name:             "default start ordinal",
+			ordinals:         &appsv1.StatefulSetOrdinals{},
+			replicas:         3,
+			expectedPodNames: []string{"sts-0", "sts-1", "sts-2"},
+		},
+		{
+			name: "start ordinal 4",
+			ordinals: &appsv1.StatefulSetOrdinals{
+				Start: 4,
+			},
+			replicas:         4,
+			expectedPodNames: []string{"sts-4", "sts-5", "sts-6", "sts-7"},
+		},
+		{
+			name: "start ordinal 5",
+			ordinals: &appsv1.StatefulSetOrdinals{
+				Start: 2,
+			},
+			replicas:         7,
+			expectedPodNames: []string{"sts-2", "sts-3", "sts-4", "sts-5", "sts-6", "sts-7", "sts-8"},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.StatefulSetStartOrdinal, true)()
+			closeFn, rm, informers, c := scSetup(t)
```

Comment:
can you check how long it takes to run these tests, and if is considerable to see if we can reuse the same setup for all the tests using namespaces for avoid test pollution?

Reply:
Thanks @aojea, what's the best way to validate the run-time of a set of tests?

Reply:
Thanks, was able to isolate these tests with:

```
go test ./test/integration/statefulset -v -run 'TestStatefulSetStartOrdinal'
```

This reduces the test run time time from 20s -> ~5s